src/ucp/api/ucp.h

```diff
@@ -1416,6 +1417,15 @@ typedef struct {
      * which means the memory type will be detected internally.
      */
     ucs_memory_type_t memory_type;
+
+    /**
+     * Pointer to the information where received data details are stored
+     * in case of immediate completion of receive operation.
+     */
+    union {
+        size_t              *length;
```

Comment:
Do we need `uint32_t mask` here to show which fields were set by UCP?
e.g. I'd not expect that `tag` is updated by UCP AM API, but a user could specify it (e.g. user shares the same request params for TAG and AM operations)

Reply:
i do not think so. This is somewhat similar to `cb` field of this struct

Reply:
Ah, this is union
I see